concordium-consensus/src/Concordium/GlobalState/Persistent/Accounts.hs

```diff
@@ -69,96 +122,111 @@ import Concordium.Types.HashableTo
-        (pMap, accountMap') <- storeUpdate accountMap
         (pTable, accountTable') <- storeUpdate accountTable
         (pRegIdHistory, regIdHistory') <- storeUpdate accountRegIdHistory
+        case accountDiffMap of
+            Nothing -> return ()
+            Just accDiffMap -> LMDBAccountMap.insert (Map.toList $ DiffMap.flatten accDiffMap)
         let newAccounts =
                 Accounts
-                    { accountMap = accountMap',
-                      accountTable = accountTable',
-                      accountRegIdHistory = regIdHistory'
+                    { accountTable = accountTable',
+                      accountRegIdHistory = regIdHistory',
+                      -- The difference map is set to @Nothing@ as any potential new accounts have been written to the
+                      -- lmdb account map.
+                      accountDiffMap = Nothing
                     }
-        return (pMap >> pTable >> pRegIdHistory, newAccounts)
+
+        -- put an empty 'OldMap.PersistentAccountMap'.
+        -- In earlier versions of the node the above mentioned account map was used,
+        -- but this is now superseded by the 'LMDBAccountMap.MonadAccountMapStore'.
+        -- We put this (0 :: Int) here to remain backwards compatible as this simply indicates an empty map.
+        -- This should be revised as part of a future protocol update when the database layout can be changed.
+        return (put (0 :: Int) >> pTable >> pRegIdHistory, newAccounts)
     load = do
-        maccountMap <- load
+        -- load the persistent account map and throw it away. We always put an empty one in,
+        -- but that has not always been the case. But the 'OldMap.PersistentAccountMap' is now superseded by
+        -- the LMDBAccountMap.MonadAccountMapStore.
+        -- This should be revised as part of a future protocol update when the database layout can be changed.
+        void (load :: Get (m (OldMap.PersistentAccountMap pv)))
         maccountTable <- load
         mrRIH <- load
         return $ do
-            accountMap <- maccountMap
             accountTable <- maccountTable
             accountRegIdHistory <- mrRIH
+            let accountDiffMap = Nothing
             return $ Accounts{..}
 
 instance (SupportsPersistentAccount pv m, av ~ AccountVersionFor pv) => Cacheable1 m (Accounts pv) (PersistentAccount av) where
     liftCache cch accts@Accounts{..} = do
-        acctMap <- cache accountMap
         acctTable <- liftCache (liftCache @_ @(HashedCachedRef (AccountCache av) (PersistentAccount av)) cch) accountTable
         return
-            accts
-                { accountMap = acctMap,
-                  accountTable = acctTable
-                }
+            accts{accountTable = acctTable}
 
--- | An 'Accounts' with no accounts.
-emptyAccounts :: Accounts pv
-emptyAccounts = Accounts AccountMap.empty L.empty Trie.empty
+emptyAccounts :: Maybe DiffMap.DifferenceMap -> Accounts pv
```

Comment:
**Suggestion (non-blocking):** Consider removing the parameter.

I think this is only ever called with `Nothing` as the first argument.